terraform-providers/terraform-provider-vcd
# Upgrade tests

* Add upgrade test environment
    * Add a `PREVIOUS_VERSION` file, containing the version from which  we want to test the upgrade
    * Add script `test-upgrade.sh` to drive upgrade operations
    * Modify `test-binary.sh` to support upgrade ops
    * Add command `make test-upgrade`

* Remove interpolation-only expressions from tests as they are causing a huge amount of warning text during binary tests.

How to run the upgrade tests:
1. Make sure you are using `terraform` binary 0.12.13. Using the latest version will blow your screen with lots of warning about interpolation-only expressions. Although these expressions are fixed in the current release, part of the test will run using code from the previous release.
2. Make sure the branch doesn't have any uncommitted files.
3. Run `make test-upgrade`
